CONTRIBUTING.md

```diff
@@ -0,0 +1,210 @@
+
+To check the formatting run:
+
+```bash
+cargo fmt --check
+```
+
+And to directly apply
+
+```bash
+cargo fmt
+```
+The formatting is also checked in the CI and a prerequisite for merging additional or changed code. 
+
+## Lints
+
+For linting `clippy` is used.
+Because the app needs to be built through meson, there is a meson target the print lints available:
+
+For the UI
+
+```bash
+meson compile ui-cargo-clippy -C _mesonbuild
+```
+
+For the CLI
+
+```bash
+meson compile cli-cargo-clippy -C _mesonbuild
+```
+
+If consciously considered clippy warnings can also disabled in code by using `#[allow(clippy::<lint-name>)].
+However this must be justified when getting new code in.
+
+## Pre-Commit hooks
+
+Per default on an initial build git pre-commit hooks are installed to ensure the outlined style consistency.
+Check out [pre-commit.hook](hooks/pre-commit.hook) to see what the hook will do in detail.
+
+# Tests
+
+## Unit Tests
+
+Some unit tests are specified throughout the codebase.
+
+To run them, execute:
+
+```bash
+meson compile cargo-test -C _mesonbuild
+```
+
+Adding tests can be done like in any other rust crate by adding `tests` modules
+and `#[test]` cases directly in code in the same files where the code that should be tested is written.
+
+## Data / Package File Validation
+
+To check the style and correctness of other data/auxiliary files like the `.desktop`
+or `metainfo.xml` AppData definition file execute:
+
+```bash
+meson test -C _mesonbuild
+```
+
+# Contributing Code
+
+All code additions should go through a PR->Review cycle.
+The core maintainers can also push directly to main but should only do that in case of
+trivial changes and fixes.
+
+The CI must run successfully in a opened PR to get it merged.
+Ideally the optional lint step does not report any warnings.
+But because new lints can appear on new clippy versions this is not mandatory.
+
+Please add a short description outlining the changes and the reasons for them.
+When adding new features and/or changes in the UI some screenshots or screen captures would be nice.
+When it fixes a specific issue, the description should reference the to-be-fixed issue with `fixes #<num>`.
+
+# Build system
+
+For building the application the `cargo` calls are wrapped by meson.
+It uses a user specified build directory (e.g. `_mesonbuild`) where the build artifacts will be compiled into.
+All needed additional files needed before/after compilation are prepared and placed into it as well.
+
+# Dependencies
+
+Rust dependencies are declared in the root workspace `Cargo.toml`, or if crate-specific
+in the individual crates `Cargo.toml` configuration files.
+
+The generated `Cargo.lock` file pins the dependencies to specific versions and is checked in.
+
+All non-rust dependencies are declared in the root `meson.build` file.
+You'll see declarations for example for `glib`, `gtk4`, `poppler` and so on.
```

Comment:
Suggestion "For example, you'll find declarations for dependencies like  `glib`, `gtk4`, and `poppler`."